src/Stratis.Bitcoin/AsyncWork/IAsyncProvider.cs

```diff
@@ -117,6 +118,15 @@ public interface IAsyncProvider
         /// <param name="faultyOnly">if set to <c>true</c> dump information only for loops in faulty state.</param>
         string GetStatistics(bool faultyOnly);
 
+        /// <summary>
+        /// temporary hack to expose signals to most of the components (every component currently using asyncprovider), while we decide if we should introduce a component
```

Comment:
@MithrilMan can you remove this comment and create an issue on the board for something a core component interface (Which I am in favor of btw)

Reply:
done
https://dev.azure.com/Stratisplatformuk/StratisBitcoinFullNode/_workitems/edit/4127